### ISSUE TITLE
Fails w/out git 

### ISSUE THREAD

**Issue by yurivict** [surge-synthesizer/surge#2681]
When git returns an error code configure prints this:

```
CMake Error at cmake/versiontools.cmake:66 (if):
  if given arguments:

    "STREQUAL" "main"

  Unknown arguments specified
```

Both situations when git is missing and when git fails should be handled gracefully.

**Comment by baconpaul** [comment]
Morning @yurivict 

Can I please ask that when you file bug reports you share the version of the code you are running?

Current state of surge is

1. If you checkout with git you need git through the entire build process
2. If you download the nightly tarball, the git information is stripped and cached and works
3. If you download the 1.7.1 tarball, the git information is missing and will fail.

The reason I ask for version and build configuration is because the 2/3 difference was fixed at 66b0cfaed90c241810569284c2983b4c1e079aca

If you are building some other way (you checkout from git and stage your own tarball and strip git and build, say), then your build formula will need to follow the steps using the scripts in the hash above. If you are in that situation we can help you.

If you are using the mechanism where you check out from git and then build late with git failing, we currently don't support that as you mention. Please let us know how that situation arose.

So basically: We need more details than your bug report.

Thank you
